src/AuthorizedSocketConnection.js

```diff
@@ -81,21 +81,25 @@ export default class AuthorizedSocketConnection<TContext, TCredentials> {
     this.socket.emit('app_error', error);
   }
 
-  isAuthorized = (data: any) => {
+  isAuthorized = (
+    data: any,
+    hasPermission: (data: any, credentials: TCredentials) => boolean,
+  ) => {
     const credentials = this.config.credentialsManager.getCredentials();
-    const isAuthorized =
-      !!credentials && this.config.hasPermission(data, credentials);
-
+    const isAuthorized = !!credentials && hasPermission(data, credentials);
     if (!isAuthorized) {
       this.log('info', 'unauthorized', {
         payload: data,
         credentials,
       });
     }
-
     return isAuthorized;
   };
 
+  hasPermission = (data: any, credentials: any) => {
+    return !!credentials && this.config.hasPermission(data, credentials);
```

Comment:
```suggestion
    return this.config.hasPermission(data, credentials);
```

the `!!credentials` check is already done above